columbo/_interaction.py

```diff
@@ -39,60 +39,121 @@ def _or_default(value, default: T) -> T:
     return default if isinstance(value, _Sentinel) else value
 
 
-class Echo:
-    """Display a message to the user."""
+class Displayable(ABC):
+    """
+    Base class for a message to the user that is displayed.
+    """
 
-    def __init__(self, message: StaticOrDynamicValue[str]) -> None:
+    def __init__(
+        self, message: StaticOrDynamicValue[str], should_ask: Optional[ShouldAsk] = None
+    ) -> None:
         """
         Initialize an instance.
 
         :param message: The message to be displayed to the user. If the value is callable, the argument passed in will
             be the answers that have been provided this far.
+        :param should_ask: If `None`, the message is displayed to the user. Otherwise, the callable will be passed the
+            answers that have been provided this far and should return `True` if the message should be displayed.
         """
         self._message = message
+        self._should_ask = should_ask
+
+    @abstractmethod
+    def display(self, answers: Answers) -> None:  # pragma: no cover
+        pass
+
+    def should_ask(self, answers: Answers) -> bool:
+        """
+        Should the user be displayed this message.
+
+        :param answers: The answers that have been provided this far.
+        :return: `True` if this message should be displayed
+        """
+        if self._should_ask is None:
+            return True
+        if callable(self._should_ask):
+            return self._should_ask(answers)
+        raise ValueError(f"Invalid value for should_ask: {self._should_ask}")
```

Comment:
Minor: Instead of duplicating this functionality, can you extract it out into a private function used by `Displayable` & `Question`?